common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1121,10 +1121,12 @@ public boolean matches(String expr) {
      */
     @Override
     public int hashCode() {
-        if (hash == 0) {
-            hash = PlatformDependent.hashCodeAscii(value, offset, length);
+        int h = hash;
```

Comment:
is the temporary variable change necessary? The member variable `hash` is set as a result of the computation, and not used during the computation ... so I wouldn't expect this to have any impact.